src/EmuFs.cc

```diff
@@ -86,14 +102,21 @@ void EmuFile::update(dev_t device, ino_t inode, uint64_t size) {
        << "-inode-" << orig_inode << "-" << path_tag;
   string real_name = name.str().substr(0, 255);
 
-  ScopedFd fd =
-      open(real_name.c_str(), O_CREAT | O_EXCL | O_RDWR | O_CLOEXEC, 0600);
+  int res = memfd_create(real_name.c_str(), MFD_CLOEXEC);
```

Comment:
Can you confirm that you really need this in practice? I'd prefer to not add another testing path without a confirmed need.